auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,129 @@
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Filter} that replaces basic auth with a bearer token.
+ *
+ * <p>It assumes that the bearer token is held in the password field of the basic auth credentials, and that it is
+ * base-64 encoded.
+ */
+public class BasicAuthToBearerTokenFilter implements Filter {
+    private static final Logger log = LoggerFactory.getLogger(BasicAuthToBearerTokenFilter.class);
+
+    private static final String BASIC_AUTH_STR = "Basic";
+    private static final BaseEncoding BASE_64_ENCODING = BaseEncoding.base64Url();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        ServletRequest updatedRequest;
+        try {
+            updatedRequest = getRequestWithToken(request);
+        } catch (IllegalArgumentException e) {
+            updatedRequest = request;
+        }
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest getRequestWithToken(ServletRequest request) {
+        if (request instanceof HttpServletRequest) {
+            HttpServletRequest httpRequest = (HttpServletRequest) request;
+            String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            return getRequestWithTokenFromRawAuthHeader(rawAuthHeader, httpRequest);
+        } else {
+            String message = "Request is not an HttpServletRequest.";
+            log.warn(message);
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromRawAuthHeader(String rawAuthHeader, HttpServletRequest request) {
+        Preconditions.checkNotNull(rawAuthHeader, "null rawAuthHeader");
+        if (isBasicAuth(rawAuthHeader)) {
+            AuthHeader authHeader = base64DecodePassword(rawAuthHeader);
+            return getRequestWithTokenFromAuthHeader(authHeader, request);
+        } else {
+            String message = "Auth header is not basic auth.";
+            log.warn(message);
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromAuthHeader(final AuthHeader authHeader, HttpServletRequest request) {
+        return new HttpServletRequestWrapper(request) {
+            @Override
+            public String getHeader(String name) {
+                if (Objects.equals(name, HttpHeaders.AUTHORIZATION)) {
```

Comment:
Using `Objects` not `Object`.

https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals(java.lang.Object,%20java.lang.Object